### PR TITLE
Add AdmissionController interface

### PR DESCRIPTION
**Pull request by mitalawachat**
Signed-off-by: Mital Awachat <awachatm@amazon.com>

### Description
This pull request serves as a starting point for Admission Control. In this pull request we are adding `AdmissionController` interface which will be fulfilled by concrete implementation in subsequent pull requests as mentioned in design proposal given below.
 
### Issues Resolved
Feature Proposal: https://github.com/opensearch-project/OpenSearch/issues/1144
Design Proposal: https://github.com/opensearch-project/OpenSearch/issues/3400
 
### Check List
- [-] New functionality includes testing.
  - [-] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/common/throttle/AdmissionController.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    /**
+     * Decrement the tracking-object with provided value and do not apply the admission control.
+     * Mostly applicable while remitting the quota.
+     *
+     * @param count value to decrement the resource tracking-object with.
+     * @return count/value released to the resource tracking object.
+     */
+    long release(long count);
+
```
Comment (Bukhtawar) [inline server/src/main/java/org/opensearch/common/throttle/AdmissionController.java:46]: These are internal working of an Admission controller, so doesn't add value exposing them

Comment (mitalawachat) [reply server/src/main/java/org/opensearch/common/throttle/AdmissionController.java:46]: Hi @Bukhtawar,
Thanks for reviewing pull request.

I was planning to create `AdmissionControlHandler`, a netty handler that extends `ChannelDuplexHandler`.

* `acquire` function would be used from `channelRead` function of `AdmissionControlHandler`
* `release` function would be used from `write` function of `AdmissionControlHandler`

Let me know your thoughts about it.

Comment (Bukhtawar) [reply server/src/main/java/org/opensearch/common/throttle/AdmissionController.java:46]: Let me rephrase it on why I call them internal working
For a particular request, release cannot release more than what it acquired. Recommend having a method addBytesAndMaybeBreak

Comment (mitalawachat) [reply server/src/main/java/org/opensearch/common/throttle/AdmissionController.java]: Thanks @Bukhtawar. Instead of separate function for `acquire` and `release`, I've added  `addBytesAndMaybeBreak`  function which returns `Releasable` which would be used to release tokens which were allocated to a particular request.
